src/encrypt.ts

```diff
@@ -1,189 +1,45 @@
 // external modules
-import { createCipheriv } from 'crypto-browserify';
+import { createEncryptionStream } from '@transcend-io/encrypt-web-streams';
 
 // local
-import { CipherGCM } from 'crypto';
-import toBuffer from 'typedarray-to-buffer';
-import { TransformStream } from './streams';
-import {
-  PenumbraDecryptionInfo,
-  PenumbraEncryptedFile,
-  PenumbraEncryptionOptions,
-  PenumbraFileWithID,
-} from './types';
+import type { JobID } from './types';
 
 // utils
-import { emitProgress, toBuff, emitJobCompletion } from './utils';
-import { logger } from './logger';
-import { PenumbraError } from './error';
-import emitError from './utils/emitError';
+import { emitJobProgress, emitJobCompletion } from './utils';
 
 /**
- * Encrypts a readable stream
- * @param jobID - Job ID
- * @param rs - A readable stream of encrypted data
- * @param cipher - The crypto module's cipher
+ * Encrypts a readable stream with an event emitter
+ * @param id - Job ID
+ * @param readableStream - A readable stream of plaintext data
  * @param contentLength - The content length of the file, in bytes
- * @param key - Decryption key Buffer
- * @param iv - Decryption IV Buffer
- * @returns A readable stream of decrypted data
+ * @param key - Encryption key Buffer
+ * @param iv - Encryption IV Buffer
+ * @returns A readable stream of encrypted data
  */
-export function encryptStream(
-  jobID: number,
-  rs: ReadableStream,
-  cipher: CipherGCM,
-  contentLength: number,
-  key: Buffer,
-  iv: Buffer,
+export function startEncryptionStreamWithEmitter(
+  id: JobID,
+  readableStream: ReadableStream,
+  contentLength: number | null,
+  key: Uint8Array,
+  iv: Uint8Array,
 ): ReadableStream {
```

Comment:
Super optional, but it might be worthwhile to turn this into a single object argument?